deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -128,6 +128,30 @@ def search(
     ):
         """DeepLakeVectorStore search method that combines embedding search, metadata search, and custom TQL search.
 
+        Examples:
+            >>> # Search using an embedding function and prompt
+            >>> data = vector_store.search(
+            >>>        prompt = "What does this chatbot do?",
+            >>>        embedding_function = <your_embedding_function>,
+            >>>        exec_option = <preferred_exec_option>,
+            >>> )
+            >>> # Search using an embedding
+            >>> data = vector_store.search(
+            >>>        embedding = <your_embedding>,
+            >>>        exec_option = <preferred_exec_option>,
+            >>> )
+            >>> # Add a filter to your search
+            >>> data = vector_store.search(
+            >>>        embedding = <your_embedding>,
+            >>>        exec_option = <preferred_exec_option>,
+            >>>        filter = {"json_tensor_name": {"key: value"}, "json_tensor_name_2": {"key_2: value_2"},...}, # Only valid for exec_option = "python"
```

Comment:
the problem with this type of query is that is that you can't do inequality conditions. The reason I know the limitations because I implemented it in langchain.

Reply:
@davidbuniat Yup. We added is just to help us out with langchain integration , but I agree that it's very rigid.

Reply:
{"json_tensor_name": {"key: value"}, "json_tensor_name_2": {"key_2: value_2"},...}, -> also do you mean

{"json_tensor_name": {"key": "value"}, "json_tensor_name_2": {"key_2": "value_2"},...}

Reply:
Hmm, this is the same as what's above?